main.py

```diff
@@ -112,18 +112,17 @@ def main():
 
     while True:
         # Messages sent to main process for handling
-
-        # WS Commands
-        while not ws_commands.empty():
-            try:
-                parse_ws_command(ws_commands.get(), serial_ws_commands, telemetry_ws_commands)
-            except ShutdownException:
-                logger.info("Ground Station shutting down...")
-                serial.terminate()
-                telemetry.terminate()
-                websocket.terminate()
-                logger.info("Ground Station shutdown.")
-                exit(0)
+        try:
+            # WS Commands
+            command = ws_commands.get()
```

Comment:
It is so sad that the fix to this was just removing a `while` loop; we really should have thought of that. Thank you!

Reply:
The actual item is that `ws_commands.get()` blocks the thread until it has something to process, but it's not easy to see if you're not familiar with it :100: